org.eclipse.jgit.test/tst/org/eclipse/jgit/api/DiffCommandTest.java

```diff
@@ -241,6 +242,31 @@ public void testNoOutputStreamSet() throws Exception {
 		assertEquals("test.txt", diff.getNewPath());
 	}
 
+	@Test
+	/**
+	 * Setting a delta filter Pattern (applies to any file) with regex that matches the
+	 * change: diff skipped.
+	 */
+	public void testDiffModified_anyDeltaFilter() throws Exception {
+		write(new File(db.getWorkTree(), "test.txt"), "test");
+		File folder = new File(db.getWorkTree(), "folder");
+		folder.mkdir();
+		write(new File(folder, "folder.txt"), "folder");
+		Git git = new Git(db);
+		git.add().addFilepattern(".").call();
+		git.commit().setMessage("Initial commit").call();
+		write(new File(folder, "folder.txt"), "folderchange");
+
+		OutputStream out = new ByteArrayOutputStream();
+		Pattern deltaFilterPattern = Pattern.compile("change");
+		List<DiffEntry> entries = git.diff().setDeltaFilterPattern(deltaFilterPattern).setOutputStream(out).call();
+		assertEquals(0, entries.size());
+
+		String actual = out.toString();
+		String expected = "";
+		assertEquals(expected.toString(), actual);
```

Comment:
Make these 3 a single line - assertEquals("", out.toString());

Reply:
fixed it